extensions/keycloak-admin-client-reactive/runtime/src/main/java/io/quarkus/keycloak/admin/client/reactive/Keycloak.java

```diff
@@ -0,0 +1,157 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.quarkus.keycloak.admin.client.reactive;
+
+import java.net.URI;
+
+import javax.ws.rs.client.Client;
+
+import org.jboss.resteasy.reactive.client.impl.ClientBuilderImpl;
+import org.jboss.resteasy.reactive.client.impl.WebTargetImpl;
+import org.jboss.resteasy.reactive.server.jackson.JacksonBasicMessageBodyReader;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.quarkus.arc.Arc;
+import io.quarkus.arc.ArcContainer;
+import io.quarkus.arc.InstanceHandle;
+import io.quarkus.keycloak.admin.client.reactive.resource.BearerAuthFilter;
+import io.quarkus.keycloak.admin.client.reactive.resource.RealmResource;
+import io.quarkus.keycloak.admin.client.reactive.resource.RealmsResource;
+import io.quarkus.keycloak.admin.client.reactive.resource.ServerInfoResource;
+import io.quarkus.keycloak.admin.client.reactive.token.TokenManager;
+import io.quarkus.rest.client.reactive.jackson.runtime.serialisers.ClientJacksonMessageBodyWriter;
+
+/**
+ * Provides a Keycloak client.
+ * To customize the underling client, use a {@link KeycloakBuilder} to create a Keycloak client.
+ *
+ * @author rodrigo.sasaki@icarros.com.br
+ * @see KeycloakBuilder
+ */
+public class Keycloak implements AutoCloseable {
+    private final TokenManager tokenManager;
+    private final String authToken;
+    private final WebTargetImpl target;
+    private final Client client;
+    private volatile boolean closed = false;
+
+    Keycloak(String serverUrl, String realm, String username, String password, String clientId, String clientSecret,
+            String grantType, String authtoken) {
+        Config config = new Config(serverUrl, realm, username, password, clientId, clientSecret, grantType);
+        ClientBuilderImpl clientBuilder = new ClientBuilderImpl();
+        clientBuilder = registerJacksonProviders(clientBuilder);
+        client = clientBuilder.build();
+        authToken = authtoken;
+        tokenManager = authtoken == null ? new TokenManager(config, client) : null;
+
+        target = (WebTargetImpl) client.target(config.getServerUrl());
+        target.register(newAuthFilter());
+    }
+
+    // this code is much more complicated than expected because it needs to handle various permutations
+    // where beans may or may not exist
+    private ClientBuilderImpl registerJacksonProviders(ClientBuilderImpl clientBuilder) {
+        ArcContainer arcContainer = Arc.container();
+        if (arcContainer == null) {
```

Comment:
when can arc container be null?

Reply:
Initially I tried to make it work it tests environments as well, but that didn't pan out. So indeed this check could be removed.